test-js/unit/pentaho/data/filter/And.Spec.js

```diff
@@ -0,0 +1,183 @@
+      });
+    }); // #and
+
+    describe("#or ", function() {
+      it("should return an Or.", function() {
+        var result = filter.or(inStock);
+        expect(result.type).toBe("$or");
+      });
+    }); // #or
+
+    describe("#invert ", function() {
+      it("should return a simplified filter, with the negated terms next to the leafs", function() {
+        var result = filter.invert();
+
+        expect(result.type).toBe("$or");
+        expect(result.operands.length).toBe(2);
+        expect(result.operands[0].type).toBe("$not");
+        expect(result.operands[0].operand.type).toBe("$in");
+        expect(result.operands[1].type).toBe("$not");
+        expect(result.operands[1].operand.type).toBe("$eq");
+      });
+    }); // #invert
+
+
+    describe("#contains", function() {
+      it("should return `true` if a given `element` belongs to the dataset.", function() {
+        var element = new Element(data, 0);
+        expect(filter.contains(element)).toBe(true);
+      });
+
+      it("should return `false` if a given `element` does not belong to the dataset.", function() {
+        var element = new Element(data, 3);
+        expect(filter.contains(element)).toBe(false);
+      });
+
+      it("should return `true` if the filter has no operands.", function() {
+        var filter = new And();
+        var element = new Element(data, 0);
+        expect(filter.contains(element)).toBe(true);
+      });
+
+
+      it("should perform an AND.", function() {
+        [0].forEach(function(rowIdx) {
+          var elementIn = new Element(data, rowIdx);
+          expect(filter.contains(elementIn)).toBe(true);
+        });
+
+        [1, 2, 3, 4, 5, 6].forEach(function(rowIdx) {
+          var elementOut = new Element(data, rowIdx);
+          expect(filter.contains(elementOut)).toBe(false);
+        });
+      });
+
+      it("should be commutative.", function() {
+        var filter2 = new And([inStock, sales12k]);
+
+        for(var k = 0; k < 7; k++) {
+          var element = new Element(data, k);
+          expect(filter.contains(element)).toBe(filter2.contains(element));
+        }
+      });
+    }); // #contains
+
+    describe("#apply", function() {
+      it("should perform an AND.", function() {
+        var filteredData = filter.apply(data);
+
+        expect(filteredData.getNumberOfRows()).toBe(1);
+        expect(filteredData.getValue(0, 0)).toBe("A");
+      });
+
+      it("should be commutative", function() {
+        var data1 = filter.apply(data);
+
+        var filter2 = new And([inStock, sales12k]);
+        var data2 = filter2.apply(data);
+
+        expect(data1.getNumberOfRows()).toBe(1);
+        expect(data1.getNumberOfRows()).toBe(data2.getNumberOfRows());
+        expect(data1.getValue(0, 0)).toBe(data2.getValue(0, 0));
+
+      });
+    }); // #apply
+
+    describe("#toSpec", function() {
+      it("should return a JSON.", function() {
+        expect(filter.toSpec()).toEqual({
+          "$and": [
```

Comment:
This is a very interesting remark that optimizes the syntax, but if you don't mind I'll ignore it for now, as I think we must first decide if this .toSpec format will be kept.